pkg/gather/clusterconfig/clusterconfig.go

```diff
@@ -237,22 +237,41 @@ func GatherContainerImages(i *Gatherer) func() ([]record.Record, []error) {
 				if strings.HasPrefix(pod.Namespace, "openshift") && hasContainerInCrashloop(podPtr) {
 					records = append(records, record.Record{Name: fmt.Sprintf("config/pod/%s/%s", pod.Namespace, pod.Name), Item: PodAnonymizer{podPtr}})
 				} else if pod.Status.Phase == corev1.PodRunning {
-					for _, container := range pod.Spec.Containers {
-						imageURL := container.Image
-						urlHost, err := forceParseURLHost(imageURL)
-						if err != nil {
-							klog.Errorf("unable to parse container image URL: %v", err)
-							continue
-						}
-						if imageHostRegex.MatchString(urlHost) {
-							imgIndex, ok := image2idx[imageURL]
-							if !ok {
-								imgIndex = contInfo.Images.Add(imageURL)
-								image2idx[imageURL] = imgIndex
+					gatherImages := func(containers []corev1.ContainerStatus) {
+						for _, container := range containers {
+							dockerRef, err := reference.Parse(container.Image)
+							if err != nil {
+								klog.Errorf("unable to parse container image specification: %v", err)
+								gatherErrors = append(gatherErrors, err)
```

Comment:
I think I would ignore parsing errors at all and return nil instead down there and return only successfully collected list (the non nill error would affect status of operator I think).

Reply:
Okay, I remember we talked about this, but I forgot to implement it. It's a fair point.